File: app/src/main/java/com/example/notes/note/papers/paper/messages/message/scraps/scrap/letters/letter/memoirs/memoir/NoteContentFragment.java
package com.example.notes.note.papers.paper.messages.message.scraps.scrap.letters.letter.memoirs.memoir;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.fragment.app.Fragment;

public class NoteContentFragment extends Fragment {

    private static final String DATA_KEY = "KEY";

    private NoteData noteData;

    public static NoteContentFragment newInstance(NoteData noteData) {
        NoteContentFragment fragment = new NoteContentFragment();
        Bundle bundle = new Bundle();
        bundle.putParcelable(DATA_KEY, noteData);
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            this.noteData = getArguments().getParcelable(DATA_KEY);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_note_content, container, false);
        TextView name = view.findViewById(R.id.name);
        TextView noteText = view.findViewById(R.id.noteText);

        name.setText(this.noteData.getName());
        noteText.setText(this.noteData.getNoteContent());

        return view;
    }

}

File: app/src/main/java/com/github/tomasSt365/notes/ui/main/NoteContentFragment.kt
package com.github.tomasSt365.notes.ui.main

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.github.tomasSt365.notes.R
import com.github.tomasSt365.notes.data.NoteData

class NoteContentFragment : Fragment() {
    private var currentNote: NoteData? = null

    @SuppressLint("UseRequireInsteadOfGet")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        currentNote = if (arguments != null) {
            arguments!!.getParcelable(DATA_KEY)
        } else {
            NoteData.Builder().build()
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_note_content, container, false)
        val name = view.findViewById<TextView>(R.id.noteTitle)
        val noteText = view.findViewById<TextView>(R.id.noteContent)
        name.text = currentNote!!.title
        noteText.text = currentNote!!.noteContent
        return view
    }

    companion object {
        private const val DATA_KEY = "KEY"
        fun newInstance(noteData: NoteData?): NoteContentFragment {
            val fragment = NoteContentFragment()
            val bundle = Bundle()
            bundle.putParcelable(DATA_KEY, noteData)
            fragment.arguments = bundle
            return fragment
        }
    }
}